sdk/attestation/azure-security-attestation/samples/async_samples/sample_attest_enclave_async.py

```diff
@@ -0,0 +1,251 @@
+        collateral. This can be extremely helpful in debugging attestation policies.
+        """
+        write_banner("attest_open_enclave_with_draft_policy")
+        oe_report = base64.urlsafe_b64decode(sample_open_enclave_report)
+        runtime_data = base64.urlsafe_b64decode(sample_runtime_data)
+
+        # [START attest_open_enclave_shared_draft]
+        draft_policy="""
+        version= 1.0;
+        authorizationrules
+        {
+            [ type=="x-ms-sgx-is-debuggable", value==false ] &&
+            [ type=="x-ms-sgx-product-id", value==1 ] &&
+            [ type=="x-ms-sgx-svn", value>= 0 ] &&
+            [ type=="x-ms-sgx-mrsigner", value=="2c1a44952ae8207135c6c29b75b8c029372ee94b677e15c20bd42340f10d41aa"]
+                => permit();
+        };
+        issuancerules {
+            c:[type=="x-ms-sgx-mrsigner"] => issue(type="My-MrSigner", value=c.value);
+        };
+        """
+        print('Attest Open enclave using ', self.shared_url)
+        print('Using draft policy:', draft_policy)
+        async with self._create_client(self.shared_url) as attest_client:
+            response = await attest_client.attest_open_enclave(
+                oe_report, runtime_data=AttestationData(runtime_data, is_json=False),
+                draft_policy=draft_policy)
+
+        print("Token algorithm", response.token.algorithm)
+        print("Issuer of token is: ", response.value.issuer)
+        # [END attest_open_enclave_shared_draft]
+
+    async def attest_open_enclave_with_draft_failing_policy(self):
+        """
+        Sets a draft policy which is guaranteed to fail attestation with the
+        sample ctest collateral to show how to manage attestation failures.
+        """
+        write_banner("attest_open_enclave_with_draft_failing_policy")
+        oe_report = base64.urlsafe_b64decode(sample_open_enclave_report)
+        runtime_data = base64.urlsafe_b64decode(sample_runtime_data)
+
+        draft_policy="""
+version= 1.0;
+authorizationrules
+{
+    [ type=="x-ms-sgx-is-debuggable", value == false] => deny();
+    [ type=="x-ms-sgx-product-id", value==1 ] &&
+    [ type=="x-ms-sgx-svn", value>= 0 ] &&
+    [ type=="x-ms-sgx-mrsigner", value=="2c1a44952ae8207135c6c29b75b8c029372ee94b677e15c20bd42340f10d41aa"]
+        => permit();
+};
+issuancerules {
+    c:[type=="x-ms-sgx-mrsigner"] => issue(type="My-MrSigner", value=c.value);
+};
+"""
+
+        print('Attest Open enclave using ', self.shared_url)
+        print('Using draft policy which will fail.:', draft_policy)
+        try:
+            async with self._create_client(self.shared_url) as attest_client:
+                await attest_client.attest_open_enclave(
+                    oe_report, runtime_data=AttestationData(runtime_data, is_json=False),
+                    draft_policy=draft_policy)
+                print("Unexpectedly passed attestation.")
+        except HttpResponseError as err:
```

Comment:
What is the expected exception? Let's change to catch only that specific exception.

Reply:
When I ran it in the debugger, the exception which was thrown from the pipeline was HttpResponseError. Is there supposed to be a better exception?

Reply:
That's fine, wanted to verify it wasn't a 409 or 404 that we have standard error types for.

Reply:
Yeah, it's a 400 error :).